src/generator/sfdx-generator.ts

```diff
@@ -103,7 +110,10 @@ export class Generator {
     const sfdxClassTemplate = _.template(templateSFDXFile);
 
     fs.writeFileSync(
-      path.resolve(this.options.outputDirectory, "generatedClient." + this.options.fileExtension),
+      path.resolve(
```

Comment:
Try using https://nodejs.org/api/path.html#path_path_format_pathobject I think that it'll be more appropriate.